### ISSUE TITLE
[Messenger / Mailer] Sending an email with binary attachment async fails

### ISSUE THREAD

**Issue by rimas-kudelis** [symfony/symfony#33913]
**Symfony version(s) affected**: 4.3.5

**Description**
When trying to send an email with a binary attachment asynchronously, Symfony encounters an exception and fails.

**How to reproduce**  
- set up async sending using the Messenger component
- attach a PDF to the email being formed. Mine is templated, and I'm attachin the PDF as a string, but I doubt it matters.

**Possible Solution**  

**Additional context**  
I tried attaching the first 20 bytes of the PDF, here's the code extract just in case:
```php
$email = new TemplatedEmail();
$fileName = sprintf('%s.pdf', $object->getName());
$email
  ->to($recipient)
  ->subject('Subject')
  ->htmlTemplate('email/template.html.twig')
  ->context([
    'some_var' => $someVar,
    'other_var' => $otherVar,
  ])
  // $pdf is a TCPDF object
  ->attach(substr($pdf->output($fileName, 'S'), 0, 20), $fileName, 'application/pdf');

$this->mailer->send($email);
```
and here's the resulting crash log from docker output:
```
php_1    | 2019-10-08T14:59:19+00:00 [debug] INSERT INTO messenger_messages (body, headers, queue_name, created_at, available_at) VALUES(?, ?, ?, ?, ?)
php_1    | 2019-10-08T14:59:19+00:00 [critical] Uncaught PHP Exception Symfony\Component\Messenger\Exception\HandlerFailedException: "An exception occurred while executing 'INSERT INTO messenger_messages (body, headers, queue_name, created_at, available_at) VALUES(?, ?, ?, ?, ?)' with params ["\x4f\x3a\x33\x36\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x45\x6e\x76\x65\x6c\x6f\x70\x65\x5c\x22\x3a\x32\x3a\x7b\x73\x3a\x34\x34\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x45\x6e\x76\x65\x6c\x6f\x70\x65\x5c\x30\x73\x74\x61\x6d\x70\x73\x5c\x22\x3b\x61\x3a\x32\x3a\x7b\x73\x3a\x34\x36\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x42\x75\x73\x4e\x61\x6d\x65\x53\x74\x61\x6d\x70\x5c\x22\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x34\x36\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x42\x75\x73\x4e\x61\x6d\x65\x53\x74\x61\x6d\x70\x5c\x22\x3a\x31\x3a\x7b\x73\x3a\x35\x35\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x42\x75\x73\x4e\x61\x6d\x65\x53\x74\x61\x6d\x70\x5c\x30\x62\x75\x73\x4e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x32\x31\x3a\x5c\x22\x6d\x65\x73\x73\x65\x6e\x67\x65\x72\x2e\x62\x75\x73\x2e\x64\x65\x66\x61\x75\x6c\x74\x5c\x22\x3b\x7d\x7d\x73\x3a\x34\x33\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x53\x65\x6e\x74\x53\x74\x61\x6d\x70\x5c\x22\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x34\x33\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x53\x65\x6e\x74\x53\x74\x61\x6d\x70\x5c\x22\x3a\x32\x3a\x7b\x73\x3a\x35\x36\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x53\x65\x6e\x74\x53\x74\x61\x6d\x70\x5c\x30\x73\x65\x6e\x64\x65\x72\x43\x6c\x61\x73\x73\x5c\x22\x3b\x73\x3a\x36\x34\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x54\x72\x61\x6e\x73\x70\x6f\x72\x74\x5c\x5c\x44\x6f\x63\x74\x72\x69\x6e\x65\x5c\x5c\x44\x6f\x63\x74\x72\x69\x6e\x65\x54\x72\x61\x6e\x73\x70\x6f\x72\x74\x5c\x22\x3b\x73\x3a\x35\x36\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x74\x61\x6d\x70\x5c\x5c\x53\x65\x6e\x74\x53\x74\x61\x6d\x70\x5c\x30\x73\x65\x6e\x64\x65\x72\x41\x6c\x69\x61\x73\x5c\x22\x3b\x73\x3a\x35\x3a\x5c\x22\x61\x73\x79\x6e\x63\x5c\x22\x3b\x7d\x7d\x7d\x73\x3a\x34\x35\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x45\x6e\x76\x65\x6c\x6f\x70\x65\x5c\x30\x6d\x65\x73\x73\x61\x67\x65\x5c\x22\x3b\x4f\x3a\x35\x31\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x61\x69\x6c\x65\x72\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x65\x6e\x64\x45\x6d\x61\x69\x6c\x4d\x65\x73\x73\x61\x67\x65\x5c\x22\x3a\x32\x3a\x7b\x73\x3a\x36\x30\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x61\x69\x6c\x65\x72\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x65\x6e\x64\x45\x6d\x61\x69\x6c\x4d\x65\x73\x73\x61\x67\x65\x5c\x30\x6d\x65\x73\x73\x61\x67\x65\x5c\x22\x3b\x43\x3a\x33\x39\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x42\x72\x69\x64\x67\x65\x5c\x5c\x54\x77\x69\x67\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x54\x65\x6d\x70\x6c\x61\x74\x65\x64\x45\x6d\x61\x69\x6c\x5c\x22\x3a\x31\x35\x34\x39\x3a\x7b\x61\x3a\x34\x3a\x7b\x69\x3a\x30\x3b\x73\x3a\x33\x38\x3a\x5c\x22\x65\x6d\x61\x69\x6c\x2f\x73\x68\x61\x72\x65\x5f\x73\x65\x73\x73\x69\x6f\x6e\x5f\x70\x6c\x61\x6e\x5f\x70\x64\x66\x2e\x68\x74\x6d\x6c\x2e\x74\x77\x69\x67\x5c\x22\x3b\x69\x3a\x31\x3b\x4e\x3b\x69\x3a\x32\x3b\x61\x3a\x32\x3a\x7b\x73\x3a\x31\x31\x3a\x5c\x22\x73\x68\x61\x72\x65\x72\x5f\x6e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x31\x37\x3a\x5c\x22\x49\x4c\x47\x20\x41\x64\x6d\x69\x6e\x69\x73\x74\x72\x61\x74\x6f\x72\x5c\x22\x3b\x73\x3a\x39\x3a\x5c\x22\x70\x6c\x61\x6e\x5f\x6e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x31\x35\x3a\x5c\x22\x50\x69\x72\x6d\x61\x73\x69\x73\x20\x70\x6c\x61\x6e\x61\x73\x5c\x22\x3b\x7d\x69\x3a\x33\x3b\x61\x3a\x36\x3a\x7b\x69\x3a\x30\x3b\x4e\x3b\x69\x3a\x31\x3b\x4e\x3b\x69\x3a\x32\x3b\x4e\x3b\x69\x3a\x33\x3b\x4e\x3b\x69\x3a\x34\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x61\x3a\x34\x3a\x7b\x73\x3a\x34\x3a\x5c\x22\x62\x6f\x64\x79\x5c\x22\x3b\x73\x3a\x32\x30\x3a\x5c\x22\x25\x50\x44\x46\x2d\x31\x2e\x37\x0a\x25\xe2\xe3\xcf\xd3\x0a\x37\x20\x30\x20\x6f\x5c\x22\x3b\x73\x3a\x34\x3a\x5c\x22\x6e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x31\x39\x3a\x5c\x22\x50\x69\x72\x6d\x61\x73\x69\x73\x20\x70\x6c\x61\x6e\x61\x73\x2e\x70\x64\x66\x5c\x22\x3b\x73\x3a\x31\x32\x3a\x5c\x22\x63\x6f\x6e\x74\x65\x6e\x74\x2d\x74\x79\x70\x65\x5c\x22\x3b\x73\x3a\x31\x35\x3a\x5c\x22\x61\x70\x70\x6c\x69\x63\x61\x74\x69\x6f\x6e\x2f\x70\x64\x66\x5c\x22\x3b\x73\x3a\x36\x3a\x5c\x22\x69\x6e\x6c\x69\x6e\x65\x5c\x22\x3b\x62\x3a\x30\x3b\x7d\x7d\x69\x3a\x35\x3b\x61\x3a\x32\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x33\x37\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x48\x65\x61\x64\x65\x72\x73\x5c\x22\x3a\x32\x3a\x7b\x73\x3a\x34\x36\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x48\x65\x61\x64\x65\x72\x73\x5c\x30\x68\x65\x61\x64\x65\x72\x73\x5c\x22\x3b\x61\x3a\x32\x3a\x7b\x73\x3a\x32\x3a\x5c\x22\x74\x6f\x5c\x22\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x34\x37\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x4d\x61\x69\x6c\x62\x6f\x78\x4c\x69\x73\x74\x48\x65\x61\x64\x65\x72\x5c\x22\x3a\x35\x3a\x7b\x73\x3a\x35\x38\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x4d\x61\x69\x6c\x62\x6f\x78\x4c\x69\x73\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x61\x64\x64\x72\x65\x73\x73\x65\x73\x5c\x22\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x33\x30\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x41\x64\x64\x72\x65\x73\x73\x5c\x22\x3a\x31\x3a\x7b\x73\x3a\x33\x39\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x41\x64\x64\x72\x65\x73\x73\x5c\x30\x61\x64\x64\x72\x65\x73\x73\x5c\x22\x3b\x73\x3a\x31\x37\x3a\x5c\x22\x72\x69\x6d\x61\x73\x40\x6e\x6f\x72\x64\x63\x6f\x64\x65\x2e\x69\x6f\x5c\x22\x3b\x7d\x7d\x73\x3a\x35\x30\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x32\x3a\x5c\x22\x54\x6f\x5c\x22\x3b\x73\x3a\x35\x36\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6c\x69\x6e\x65\x4c\x65\x6e\x67\x74\x68\x5c\x22\x3b\x69\x3a\x37\x36\x3b\x73\x3a\x35\x30\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x4
php_1    | 1\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6c\x61\x6e\x67\x5c\x22\x3b\x4e\x3b\x73\x3a\x35\x33\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x63\x68\x61\x72\x73\x65\x74\x5c\x22\x3b\x73\x3a\x35\x3a\x5c\x22\x75\x74\x66\x2d\x38\x5c\x22\x3b\x7d\x7d\x73\x3a\x37\x3a\x5c\x22\x73\x75\x62\x6a\x65\x63\x74\x5c\x22\x3b\x61\x3a\x31\x3a\x7b\x69\x3a\x30\x3b\x4f\x3a\x34\x38\x3a\x5c\x22\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x55\x6e\x73\x74\x72\x75\x63\x74\x75\x72\x65\x64\x48\x65\x61\x64\x65\x72\x5c\x22\x3a\x35\x3a\x7b\x73\x3a\x35\x35\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x55\x6e\x73\x74\x72\x75\x63\x74\x75\x72\x65\x64\x48\x65\x61\x64\x65\x72\x5c\x30\x76\x61\x6c\x75\x65\x5c\x22\x3b\x73\x3a\x34\x36\x3a\x5c\x22\x43\x72\x69\x63\x6b\x65\x74\x20\x70\x72\x61\x63\x74\x69\x63\x65\x20\x70\x6c\x61\x6e\x20\x68\x61\x73\x20\x62\x65\x65\x6e\x20\x73\x68\x61\x72\x65\x64\x20\x77\x69\x74\x68\x20\x79\x6f\x75\x5c\x22\x3b\x73\x3a\x35\x30\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6e\x61\x6d\x65\x5c\x22\x3b\x73\x3a\x37\x3a\x5c\x22\x53\x75\x62\x6a\x65\x63\x74\x5c\x22\x3b\x73\x3a\x35\x36\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6c\x69\x6e\x65\x4c\x65\x6e\x67\x74\x68\x5c\x22\x3b\x69\x3a\x37\x36\x3b\x73\x3a\x35\x30\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x6c\x61\x6e\x67\x5c\x22\x3b\x4e\x3b\x73\x3a\x35\x33\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x41\x62\x73\x74\x72\x61\x63\x74\x48\x65\x61\x64\x65\x72\x5c\x30\x63\x68\x61\x72\x73\x65\x74\x5c\x22\x3b\x73\x3a\x35\x3a\x5c\x22\x75\x74\x66\x2d\x38\x5c\x22\x3b\x7d\x7d\x7d\x73\x3a\x34\x39\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x69\x6d\x65\x5c\x5c\x48\x65\x61\x64\x65\x72\x5c\x5c\x48\x65\x61\x64\x65\x72\x73\x5c\x30\x6c\x69\x6e\x65\x4c\x65\x6e\x67\x74\x68\x5c\x22\x3b\x69\x3a\x37\x36\x3b\x7d\x69\x3a\x31\x3b\x4e\x3b\x7d\x7d\x7d\x7d\x73\x3a\x36\x31\x3a\x5c\x22\x5c\x30\x53\x79\x6d\x66\x6f\x6e\x79\x5c\x5c\x43\x6f\x6d\x70\x6f\x6e\x65\x6e\x74\x5c\x5c\x4d\x61\x69\x6c\x65\x72\x5c\x5c\x4d\x65\x73\x73\x65\x6e\x67\x65\x72\x5c\x5c\x53\x65\x6e\x64\x45\x6d\x61\x69\x6c\x4d\x65\x73\x73\x61\x67\x65\x5c\x30\x65\x6e\x76\x65\x6c\x6f\x70\x65\x5c\x22\x3b\x4e\x3b\x7d\x7d", "[]", "default", "2019-10-08T14:59:19", "2019-10-08T14:59:19"]:
php_1    | SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect string value: '\xE2\xE3\xCF\xD3\x0A7...' for column `icc`.`messenger_messages`.`body` at row 1" at /srv/app/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php line 82
```
I tried changing the `body` and `headers` fields of the `messenger_messages` DB table to `longblob`, but it didn't fix the issue. I suppose encoding the body in b64 or qp before attempting to store it could help.

**Comment by rimas-kudelis** [comment]
You should probably tag this with **Messenger** too. It seems Mailer simply triggers a problem in Messenger in this case, and the problem I guess is that Messenger stores serialized Message object directly into the database, but that it appears that that object might still be binary unsafe.

**Comment by rimas-kudelis** [comment]
Ahh, it seems I was wrong. I guess I only thought I changed the column type, but didn't apply the change, that's why I concluded that the column type change had no effect.
I changed it properly today and can confirm that after the change, the problem seems fixed. Like I said, the problem is actually in Messenger, not in Mailer.

**Comment by OskarStark** [comment]
I would rather write this file somewhere instead of serializing the content to the message itself, and load and attach the file when handling the message

**Comment by rimas-kudelis** [comment]
You can't guarantee that the file will still be available when and where the message will be processed.

Furthermore, if the Messenger stores serialized objects in the database, the `blob` column type is more appropriate for that than `text` anyway.

**Comment by Tobion** [comment]
What serializer are you using for the transport? The default PhpSerializer?
It's using addslashes to encode null byte added by serialize(), see https://github.com/symfony/symfony/blob/b54997d1dfbbd7afabced9dbc3b3544448f02ca9/src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php#L44
But I guess if the data itself contains other binary data, that would still pass. There are three solutions:
1. binary data is not allowed
2. encode all binary data (not just null byte) as printable text
3. persist data as binary (blob)

**Comment by rimas-kudelis** [comment]
@Tobion I guess I'm using the default serializer. The scenario to reproduce this is described in the initial comment. #33920 seems to fix this issue for me.

**Comment by toooni** [comment]
This is related to my PR https://github.com/symfony/symfony/pull/35137.
The PR is not a solution to this issue but makes the mailer throw a `TransportException` for a start if the body cannot be encoded by `json_encode`.

**Comment by cb1986ster** [comment]
On dev env I use sqlite and it work, on prod I have mysql and problem show.

**Comment by cb1986ster** [comment]
@rimas-kudelis I change my template and context to use array whit one element - array of string values and somehow it works. If you still have problem try:
```
->context(['data'=>[
    'some_var' => $someVar,
    'other_var' => $otherVar,
]])
```

**Comment by ricohumme** [comment]
This issue still exists today, unfortunately.
I also this encountered when sending a message with a binary file (pdf in this case). 
When sending it synchronously, there are no issues, but when using async, the error occurs.
After updating the column `body` to a LONGBLOB. the database was able to store the message.
Afterwards the messenger successfully consumed the message and delivered it.

**Comment by ricohumme** [comment]
As stated in https://github.com/symfony/symfony/pull/33920#issuecomment-642484180 I feel the mailer component is responsible for correctly encoding the data upon serialization. Base64 en-/decoding the binary data within an email message is the way to go as far as I can see.

Please let me know what you think.

**Comment by leonelvsc** [comment]
We're doing the same with a custom Serializer and it works with base64 encoding and text column type

```
class CustomEmailSerializer implements SerializerInterface
{

    public function decode(array $encodedEnvelope): Envelope
    {
        return unserialize(base64_decode($encodedEnvelope['body']));
    }

    public function encode(Envelope $envelope): array
    {
        $envelope = $envelope->withoutStampsOfType(NonSendableStampInterface::class);

        $body = base64_encode((serialize($envelope)));

        return [
            'body' => $body,
        ];
    }
}
```

**Comment by fabpot** [comment]
If you are using files, the issue is fixed in 5.2. Closing as I think the other issue is about how to store the payload and there is already a PR for that.
